demos/common/processing/async_pipeline.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "async_pipeline.h"
+#include <samples/args_helper.hpp>
+#include <cldnn/cldnn_config.hpp>
+#include <samples/slog.hpp>
+
+using namespace InferenceEngine;
+
+PipelineBase::PipelineBase(ModelBase* modelInstance, const CnnConfig& cnnConfig, InferenceEngine::Core* engine){
+    model.reset(modelInstance);
+
+    auto ie = engine ?
+        std::unique_ptr<InferenceEngine::Core, void(*)(InferenceEngine::Core*)>(engine, [](InferenceEngine::Core*) {}) :
+        std::unique_ptr<InferenceEngine::Core, void(*)(InferenceEngine::Core*)>(new InferenceEngine::Core, [](InferenceEngine::Core* ptr) {delete ptr; });
+
+    // --------------------------- 1. Load inference engine ------------------------------------------------
+    slog::info << "Loading Inference Engine" << slog::endl;
+
+    slog::info << "Device info: " << slog::endl;
+    std::cout << ie->GetVersions(cnnConfig.devices);
+
+    /** Load extensions for the plugin **/
+    if (!cnnConfig.cpuExtensionsPath.empty()) {
+        // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
+        IExtensionPtr extension_ptr = make_so_pointer<IExtension>(cnnConfig.cpuExtensionsPath.c_str());
+        ie->AddExtension(extension_ptr, "CPU");
+    }
+    if (!cnnConfig.clKernelsConfigPath.empty()) {
+        // clDNN Extensions are loaded from an .xml description and OpenCL kernel files
+        ie->SetConfig({ {PluginConfigParams::KEY_CONFIG_FILE, cnnConfig.clKernelsConfigPath} }, "GPU");
+    }
+
+    // --------------------------- 2. Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
+    slog::info << "Loading network files" << slog::endl;
+    /** Read network model **/
+    InferenceEngine::CNNNetwork cnnNetwork = ie->ReadNetwork(model->getModelFileName());
+    /** Set batch size to 1 **/
+    slog::info << "Batch size is forced to 1." << slog::endl;
+    cnnNetwork.setBatchSize(1);
```

Comment:
`reshape()` is better.

Reply:
that's taken from original demo code. Could you explain why it is better ?

Reply:
https://github.com/openvinotoolkit/openvino/blob/665fd1b7739914d39c52d9df9490ba7b5cd7cff4/inference-engine/include/ie_icnn_network.hpp#L119

Reply:
fixed